sample-wasm-chaincode/chaincode_example02/c/main.c

```diff
@@ -0,0 +1,313 @@
+    if (args != 1) {
+        const char *str = "ERROR! Incorrect number of arguments. Expecting 1";
+        __print_wrapper(str);
+        return -1;
+    }
+
+    //Get first parameter as account name
+    const char *accountName = malloc(24 * sizeof(char));
+    int accountNameLen = __get_parameter(0, accountName);
+
+    //delete account from state
+    int result = __delete_state(accountName, accountNameLen);
+
+    //Return error if account balance length is negative
+    if (result < 0) {
+        const char *str = "Failed to delete state";
+        __return_result_wrapper(str);
+        return -1;
+    }
+    //Return zero to show success
+    return 0;
+}
+
+
+void __print_wrapper_size(const char *str, int size) {
+    __print(str, size);
+}
+
+void __print_wrapper(const char *str) {
+    int size;
+    for (size = 0; str[size] != '\0'; ++size);
+    __print_wrapper_size(str, size);
+}
+
+void __return_result_wrapper(const char *str) {
+    int size;
+    for (size = 0; str[size] != '\0'; ++size);
+    __return_result(str, size);
+}
+
+int char2int(const char *array, int n) {
+    int counter = 0;
+    int results = 0;
+    while (1) {
+        if (array[counter] == '\0') {
+            break;
+        } else {
+            results *= 10;
+            results += (int) array[counter] - 48;
+            counter++;
+        }
+    }
+    return results;
+
+}
+
+char *int2char(int iNumber) {
+    int iNumbersCount = 0;
+    int iTmpNum = iNumber;
+    while (iTmpNum) {
+        iTmpNum /= 10;
+        iNumbersCount++;
+    }
+    char *buffer = malloc(iNumbersCount + 1);
+    for (int i = iNumbersCount - 1; i >= 0; i--) {
+        buffer[i] = (char) ((iNumber % 10) | 48);
+        iNumber /= 10;
+    }
+    buffer[iNumbersCount] = '\0';
+    return buffer;
+
+}
+
+/*int __put_state_wrapper(const char *key, const char *value) {
+    int keySize;
+    for (keySize = 0; key[keySize] != '\0'; ++keySize);
+    int valueSize;
+    for (valueSize = 0; value[valueSize] != '\0'; ++valueSize);
+    return __put_state(key, keySize, value, valueSize);
+}*/
+
+/* convert character array to integer */
+/*int char2int (char *array, int n){
+    int number = 0;
+    int mult = 1;
+
+    n = (int)n < 0 ? -n : n;       *//* quick absolute value check  *//*
+
+    *//* for each character in array *//*
+    while (n--)
+    {
+        *//* if not digit or '-', check if number > 0, break or continue *//*
+        if ((array[n] < '0' || array[n] > '9') && array[n] != '-') {
+            if (number)
+                break;
+            else
+                continue;
+        }
+
+        if (array[n] == '-') {      *//* if '-' if number, negate, break *//*
+            if (number) {
+                number = -number;
+                break;
+            }
+        }
+        else {                      *//* convert digit to numeric value   *//*
+            number += (array[n] - '0') * mult;
+            mult *= 10;
+        }
+    }
+
+    return number;
+}*/
```

Comment:
Do we need the commented out stuff?

Reply:
That's another way for converting char pointer to int. I was thinking to keep it and if we ran into some problem with current method, I will switch to this one..

Reply:
If we don't need this, I am inclined to say delete it. Though I understand you wanting to be able to keep the code around. Is it possible to keep the commented code as a branch on your fork, so you have it in case we have issues with the implementation that is not commented out?

Reply:
Removed it. I copied it to another branch